index.bs

```diff
@@ -4109,165 +4310,1257 @@ To <dfn>get the response data</dfn> given |response|:
-   production, with the <code>params</code> field set to |params|.
+  1. Append `<code>;Secure</code>` to |header value|.
 
-1. [=Emit events=] with |body| and |related browsing contexts|.
+1. If |protocol cookie| [=map/contains=] "<code>httpOnly</code>" and |protocol
+   cookie|["<code>httpOnly</code>"] is true:
 
-</div>
+  1. Append `<code>;HttpOnly</code>` to |header value|.
 
-#### The network.responseStarted Event #### {#event-network-responseStarted}
+1. If |protocol cookie| [=map/contains=] "<code>sameSite</code>":
 
-<dl>
-   <dt>Event Type</dt>
+  1. Let |attribute| be `<code>;SameSite=</code>`
+
+  1. Append [=UTF-8 encode=] |protocol cookie|["<code>sameSite</code>"] to
+     |attribute|.
+
+  1. Append |attribute| to |header value|.
+
+1. Return |header value|.
+
+</div>
+
+#### The network.UrlPattern Type #### {#type-network-UrlPattern}
+
+[=Remote end definition=]
+
+<pre class="cddl remote-cddl">
+network.UrlPattern = network.UrlPatternPattern // network.UrlPatternString
+
+network.UrlPatternPattern = {
+    type = "pattern",
+    ?protocol: text,
+    ?hostname, text,
+    ?port: text,
+    ?pathname: text,
+    ?search: text,
+}
+
+
+network.UrlPatternString = {
+    type = "string",
+    pattern: text,
+}
+
+</pre>
+
+A <code>network.UrlPattern</code> represents a pattern used for matching request
+URLs for [=network intercepts=].
+
+When URLs are matched against a <code>network.UrlPattern</code> the URL is
+parsed, and each component is comapred for equality with the corresponding field
+in the pattern, if present. Missing fields from the pattern always match.
+
+Note: This syntax is designed with future extensibility in mind. In particular
+the syntax forbids characters that are treated specially in the [[URLPattern]]
+specification. These can be escaped by prefixing them with a U+005C (\) character.
+
+<div algorithm>
+To <dfn>unescape URL pattern</dfn> given |pattern|
+
+1. Let |forbidden characters| be the [=/set=] of codepoints «U+0028 ((), U+0029
+   ()), U+002A (*), U+007B ({), U+007D (})»
+
+1. Let |result| be the empty string.
+
+1. Let |is escaped character| be false.
+
+1. For each |codepoint| in |pattern|:
+
+  1. If |is escaped character| is false:
+
+    1. If |forbidden characters| [=set/contains=] |codepoint|, return [=error=]
+       with [=error code=] [=invalid argument=].
+
+    1. If |codepoint| is U+005C (\):
+
+        1. Set |is escaped character| to true.
+
+        1. Continue.
+
+  1. Append |codepoint| to result.
+
+  1. Set |is escaped character| to false.
+
+1. Return [=success=] with data |result|.
+
+</div>
+
+<div algorithm>
+To <dfn>parse URL pattern</dfn>, given |pattern|:
+
+1. Let |has protocol| be true.
+
+1. Let |has hostname| be true.
+
+1. If |pattern| matches the <code>network.UrlPatternPattern</code> production:
+
+  1. Let |pattern url| be the empty string.
+
+    1. If |pattern| [=map/contains=] "<code>protocol</code>":
```

Comment:
The indentation seem to be totally off here. It would be good to get this fixed before we merge the PR.